index.js

```diff
@@ -0,0 +1,666 @@
+    },
+
+    //  Array#traverse :: Applicative f => Array a ~> ((a -> f b), (c -> f c)) -> f (Array b)
+    'Array#fantasy-land/traverse': function(f, of) {
+      var applicative = of([]);
+      for (var idx = this.length - 1; idx >= 0; idx -= 1) {
+        applicative = ap(map(prepend, f(this[idx])), applicative);
+      }
+      return applicative;
+    },
+
+    //  Function#toString :: Function ~> String
+    'Function#toString': function() { return String(this); },
+
+    //  Function#equals :: Function ~> Function -> Boolean
+    'Function#fantasy-land/equals': function(other) { return other === this; },
+
+    //  Function#map :: (a -> b) ~> (b -> c) -> (a -> c)
+    'Function#fantasy-land/map': function(f) {
+      var functor = this;
+      return function(x) { return f(functor(x)); };
+    },
+
+    //  Function#ap :: (a -> b -> c) ~> (a -> b) -> (a -> c)
+    'Function#fantasy-land/ap': function(f) {
+      var apply = this;
+      return function(x) { return apply(x)(f(x)); };
+    },
+
+    //  Number#toString :: Number ~> String
+    'Number#toString': function() {
+      return typeof this === 'object' ?
+        'new Number(' + Z.toString(this.valueOf()) + ')' :
+        1 / this === -Infinity ? '-0' : this.toString(10);
+    },
+
+    //  Number#equals :: Number ~> Number -> Boolean
+    'Number#fantasy-land/equals': function(other) {
+      return typeof other === typeof this &&      // 42 =/= new Number(42)
+             isNaN(this) ?
+               isNaN(other) :                     // NaN === NaN
+               Number(this) === 0 ?
+                 1 / other === 1 / this :         // 0 =/= -0
+                 Number(other) === Number(this);
+    },
+
+    //  Object#toString :: StrMap a ~> String
+    'Object#toString': function() {
+      var self = this;
+      var entries = Object.keys(this).sort().map(function(k) {
+        return quote(k) + ': ' + Z.toString(self[k]);
+      });
+      return '{' + entries.join(', ') + '}';
+    },
+
+    //  Object#equals :: StrMap a ~> StrMap a -> Boolean
+    'Object#fantasy-land/equals': function(other) {
+      var self = this;
+      var keys = Object.keys(this).sort();
+      return equals(Object.keys(other).sort(), keys) &&
+             keys.every(function(k) { return equals(other[k], self[k]); });
+    },
+
+    //  Object#concat :: StrMap a ~> StrMap a
+    'Object#fantasy-land/concat': function(strMap) {
+      var result = {};
+      for (var k in this) if (has(k, this)) result[k] = this[k];
+      for (k in strMap) if (has(k, strMap)) result[k] = strMap[k];
+      return result;
+    },
+
+    //  Object#empty :: StrMap a ~> StrMap a
+    'Object#fantasy-land/empty': function() { return {}; },
+
+    //  Object#map :: StrMap a ~> (a -> b) -> StrMap b
+    'Object#fantasy-land/map': function(f) {
+      var result = {};
+      for (var k in this) if (has(k, this)) result[k] = f(this[k]);
+      return result;
+    },
+
```

Comment:
`g = a => b => c => f(a, f(b, c))`, where `f` is the reducing function `:: a -> b -> b`, has to be commutative in its first two arguments.